modules/rate_cacher/doc/rate_cacher_admin.xml

```diff
@@ -0,0 +1,636 @@
+		<para>
+		The DB URL for querying the Ratesheets used by the module
+		</para>
+		<para>
+		<emphasis>
+			Default value is <quote>NULL</quote>.
+		</emphasis>
+		</para>
+		<example>
+		<title>Setting the <varname>rates_db_url</varname> parameter</title>
+		<programlisting format="linespecific">
+...
+modparam("rate_cacher", "rates_db_url", "mysql://opensips:opensipsrw@localhost/opensips")
+...
+</programlisting>
+		</example>
+	</section>
+
+	<section id="rates_db_table" xreflabel="rates_db_table">
+		<title><varname>rates_db_table</varname> (str)</title>
+		<para>
+		The DB Table for querying the Ratesheets used by the module
+		</para>
+		<para>
+		<emphasis>
+			Default value is <quote>rc_ratesheets</quote>.
+		</emphasis>
+		</para>
+		<example>
+		<title>Setting the <varname>rates_db_table</varname> parameter</title>
+		<programlisting format="linespecific">
+...
+modparam("rate_cacher", "rates_db_table", "my_clients_view")
+...
+</programlisting>
+		</example>
+	</section>
+
+	</section>
+
+	<section id="exported_functions" xreflabel="exported_functions">
+	<title>Exported Functions</title>
+	<section id="cost_based_filtering" xreflabel="cost_based_filtering">
+		<title>
+		<function moreinfo="none">cost_based_filtering(client_id,is_wholesale,vendors_csv,dialled_no,desired_margin,out_vendor_csv)</function>
+		</title>
+		<para>
+		For a call originating from the provided Client ID, on a wholesale or retail quality, going to dialled_no, the function removes the Vendors ( from the vendor_csv list ) which do not pass the desired_margin condition, and sets the out_vendor_csv variable to the list of Vendor that meet the margin condition, while maintaining the initial order provided in the vendor_csv variable.
+		</para>
+		<para>
+		The <emphasis>client_id</emphasis> pseudo-var will hold the client_id originating this call
+		</para>
+		<para>
+		The  <emphasis>is_wholesale</emphasis> pseudo-var will contain either a 1 or a 0, depending on whether the call is wholesale or retail ( see client ratesheet provisioning ).
+		</para>
+		<para>
+		The <emphasis>vendors_csv</emphasis> pseudo-var contains a list of Vendors that need to be filtered based on the desired margin ( keep just those that match your desired percentage margin for this call )
+		</para>
+		<para>
+		The <emphasis>dialled_no</emphasis> pseudo-var contains the DNIS - the dialled number for the current call. It needs to be in E164 format, without the leading +
+		</para>
+		<para>
+		The <emphasis>desired_margin</emphasis> pseudo-var contains the minimum Integer margin that the script writer wants to achieve, based on the Client sell and Vendor buy prices. The formula used is : vendor_margin=(client_price - results[i])*100/client_price) . If the vendor_margin is higher than the desired_margin, then the Vendor is ok to use. The desired margin can be positive ( call will be profitable ) or negative ( the call will cause a loss ).
```

Comment:
15. There are quite a few inaccuracies in this one sentence. For example:

* `The desired margin can be ... negative ( the call will cause a loss )`. This is not true, I can feed the function with `-3%` and still make profit from *all* my calls. More accurate would be: `the call may cause a loss`
* `The desired margin can be positive ... or negative`. What if I set it to zero? :)

I suggest:

```
The desired margin can be positive (call must be profitable), zero (call must not cause a loss) or negative (call may cause a loss).
```